app/addons/documents/mango/components/ExecutionStats.js

```diff
@@ -0,0 +1,127 @@
+
+    return minutes + 'minute, ' + seconds + 'seconds';
+  }
+
+  getWarning(executionStats, warning) {
+    if (executionStats) {
+      // warn if many documents scanned in relation to results returned
+      if (executionStats && executionStats.results_returned) {
+        const docsExamined = executionStats.total_docs_examined || executionStats.total_quorum_docs_examined;
+        if (docsExamined / executionStats.results_returned > 10) {
+          return TOO_MANY_DOCS_SCANNED_WARNING;
+        }
+      }
+    }
+
+    return warning;
+  }
+
+  warningPopupComponent(warningText) {
+    if (!!warningText) {
+      return (<div className="warning">
+        <i className="fonticon-attention-circled"></i> {warningText}
+      </div>);
+    }
+  }
+
+  executionStatsLine(title, value, alwaysShow = false, units = "") {
+    const hasValue = value === 0 && !alwaysShow ? "false" : "true";
+    return <div data-status={hasValue}>{title + ": "}<span className="value">{value.toLocaleString()} {units}</span></div>;
+  }
+
+  executionStatsPopupComponent(executionStats) {
+    if (!executionStats) return null;
+    return (
+      <div className="execution-stats-popup-component">
+        {/* keys examined always 0 so hide it for now */}
+        {/* {this.executionStatsLine("keys examined", executionStats.total_keys_examined)} */}
+        {this.executionStatsLine("documents examined", executionStats.total_docs_examined)}
+        {this.executionStatsLine("documents examined (quorum)", executionStats.total_quorum_docs_examined)}
+        {this.executionStatsLine("results returned", executionStats.results_returned, true)}
+        {this.executionStatsLine("execution time", executionStats.execution_time_ms, false, "ms")}
+      </div>
+    );
+  }
+
+  popup(executionStats, warningText) {
+    return (
+      <Popover id="popover-trigger-hover-focus" title="Execution Statistics">
+        <div className="execution-stats-popup">
+          {this.executionStatsPopupComponent(executionStats)}
+          {this.warningPopupComponent(warningText)}
+        </div>
+      </Popover>
+    );
+  }
+
+  render() {
+    const {
+      executionStats,
+      warning
+    } = this.props;
+
+    const warningText = this.getWarning(executionStats, warning);
+
+    let warningComponent = null;
+    if (!!warningText) {
+      warningComponent = <i className="fonticon-attention-circled"></i>;
+    }
+
+    let executionStatsComponent = null;
+    if (executionStats) {
+      executionStatsComponent = (
+        <span className="execution-stats-component">Executed in {this.humanizeDuration(executionStats.execution_time_ms)}</span>
+      );
+    } else if (!!warningText) {
+      executionStatsComponent = (
+        <span className="execution-stats-component">Warning</span>
+      );
+    }
+
+    const popup = this.popup(executionStats, warningText);
+    return (
+        <OverlayTrigger trigger={['hover', 'focus']} placement="right" overlay={popup}>
```

Comment:
Can you also add click as a trigger. Just in case anyone uses this with a touch screen